### DISCUSSION TITLE
Update-failed issue.

### DISCUSSION THREAD

**Discussion by noahwseo** [community/community#112517] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi everyone,

I'm new to using GitHub for my personal blog (essentially academic resume pages). Recently, I've encountered an issue with "Update Failed." Whenever I make or modify files on certain pages and commit them for update, I receive two emails each time, indicating a failed update: "Pages build and deployment: Some jobs were not successful." Upon clicking "View workflow run," I see the following two messages. However, since I'm not well-versed in syntax, code, or programming, I'm unsure how to resolve them. Could someone please provide guidance on how to address this issue?

1) Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: peaceiris/actions-hugo@v2, actions/cache@v3, actions/configure-pages@v3. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/

2) Logging at level: debug GitHub Pages: github-pages v231 GitHub Pages: jekyll v3.9.5 Theme: jekyll-theme-primer Theme source: /usr/local/bundle/gems/jekyll-theme-primer-0.6.0 Requiring: jekyll-github-metadata Requiring: jekyll-seo-tag Requiring: jekyll-coffeescript Requiring: jekyll-commonmark-ghpages Requiring: jekyll-gist Requiring: jekyll-github-metadata Requiring: jekyll-paginate Requiring: jekyll-relative-links Requiring: jekyll-optional-front-matter Requiring: jekyll-readme-index Requiring: jekyll-default-layout Requiring: jekyll-titles-from-headings GitHub Metadata: Initializing... Source: /github/workspace/. Destination: /github/workspace/./_site Incremental build: disabled. Enable with --incremental Generating... Generating: JekyllOptionalFrontMatter::Generator finished in 0.000175957 seconds. Generating: JekyllReadmeIndex::Generator finished in 0.001983677 seconds. Generating: Jekyll::Paginate::Pagination finished in 2.745e-06 seconds. GitHub Metadata: Generating for noahwseo/noahwseo.github.io GitHub Metadata: Calling @client.pages("noahwseo/noahwseo.github.io", {}) Generating: JekyllRelativeLinks::Generator finished in 0.110481239 seconds. Generating: JekyllDefaultLayout::Generator finished in 0.000378092 seconds. Requiring: kramdown-parser-gfm Generating: JekyllTitlesFromHeadings::Generator finished in 0.008230828 seconds. Rendering: content/project/example/index.md Pre-Render Hooks: content/project/example/index.md Rendering Markup: content/project/example/index.md Rendering Layout: content/project/example/index.md Layout source: theme GitHub Metadata: Calling @client.organization("noahwseo") GitHub Metadata: Calling @client.user("noahwseo") GitHub Metadata: Calling @client.repository("noahwseo/noahwseo.github.io", {:accept=>"application/vnd.github.drax-preview+json"}) Rendering: content/project/external-project/index.md Pre-Render Hooks: content/project/external-project/index.md Rendering Markup: content/project/external-project/index.md Rendering Layout: content/project/external-project/index.md Layout source: theme Rendering: content/event/example/index.md Pre-Render Hooks: content/event/example/index.md Rendering Liquid: content/event/example/index.md Rendering Markup: content/event/example/index.md Rendering Layout: content/event/example/index.md Layout source: theme Rendering: content/post/rasch-analysis/index.md Pre-Render Hooks: content/post/rasch-analysis/index.md Rendering Markup: content/post/rasch-analysis/index.md Rendering Layout: content/post/rasch-analysis/index.md Layout source: theme Rendering: content/post/Workshop-Rasch-Analysis/index.md Pre-Render Hooks: content/post/Workshop-Rasch-Analysis/index.md Rendering Liquid: content/post/Workshop-Rasch-Analysis/index.md Rendering Markup: content/post/Workshop-Rasch-Analysis/index.md Rendering Layout: content/post/Workshop-Rasch-Analysis/index.md Layout source: theme Rendering: content/post/getting-started/index.md Pre-Render Hooks: content/post/getting-started/index.md Rendering Markup: content/post/getting-started/index.md Rendering Layout: content/post/getting-started/index.md Layout source: theme Rendering: content/slides/example/index.md Pre-Render Hooks: content/slides/example/index.md Rendering Liquid: content/slides/example/index.md github-pages 231 | Error: Liquid syntax error (line 51): Variable '{{2\left( {x + 4}' was not properly terminated with regexp: /\}\}/.

**Comment by adarsh-site** [comment]
The issue you're encountering seems to be related to the deprecation of Node.js 16 in GitHub Actions, as well as a Liquid syntax error in one of your Markdown files. Let's break down the steps you need to take to address these issues:

Node.js Version Update:
GitHub has deprecated Node.js 16 in favor of Node.js 20 for GitHub Actions. You need to update the actions in your workflow files to use Node.js 20. The specific actions mentioned in the error message are:

peaceiris/actions-hugo@v2
actions/cache@v3
actions/configure-pages@v3
You need to find these actions in your GitHub Actions workflow files (typically located in the .github/workflows directory) and update their versions to ones that support Node.js 20. You can find the updated versions in their respective repositories or documentation.

Liquid Syntax Error:
The error message indicates a Liquid syntax error in one of your Markdown files. Liquid is a templating language used in Jekyll, which is the engine behind GitHub Pages.

The specific error message is:

```
Liquid syntax error (line 51): Variable '{{2\left( {x + 4}' was not properly terminated with regexp: /}}/.
```

This suggests that there's an issue with the Liquid syntax in line 51 of one of your Markdown files. You'll need to find this file (content/slides/example/index.md according to the error message) and locate the Liquid expression causing the problem. Ensure that all Liquid expressions are properly formatted and terminated with {{ ... }}.

Once you've made these changes, commit them to your repository and push them to GitHub. GitHub Actions will then run your workflows with the updated Node.js version, and hopefully, the build will succeed without errors.
